### ISSUE TITLE
PreferKeyPaths rule prevents compilation

### ISSUE THREAD

**Issue by k-kohey** [nicklockwood/SwiftFormat#1932]
Formatting the code using version 0.55.2 results in the following uncompilable code.

```swift
// before
var hoge: [Int] {
  [0].map { $0 }
}

// after
var hoge: [Int] {
  [0].map(\.self) // Cannot convert value of type 'WritableKeyPath<_, _>' to expected argument type '(Int) throws -> Int'
}
```

Is this a bug in the formatter or is it a problem in my environment?
I am using Swift 5.10.

**Comment by nicklockwood** [comment]
Sorry, I thought that this was supposed to work now, but it seems not

**Comment by k-kohey** [comment]
I don't know why, but Swfit 6.0 can compile.

**Comment by nicklockwood** [comment]
@k-kohey fixed in 0.55.3

**Comment by Lutzifer** [comment]
@nicklockwood thanks for fixing this so fast
